log/scope_test.go

```diff
@@ -54,8 +55,8 @@ func TestBasicScopes(t *testing.T) {
 		},
 
 		{
-			f:   func() { s.Info("Hello") },
-			pat: timePattern + "\tinfo\ttestScope\tHello",
+			f:   func() { s.Info("Hello", zap.String("key", "value"), zap.Int("intkey", 123)) },
```

Comment:
what about `Info("foo", "bar", "baz")`?

Reply:
Seems like this should either work or the type should be changed on Info to only take ZapField?

Reply:
added, thanks

Reply:
so its an error? then why don't we have the old `Info(msg string, fields ...zapcore.Field)` as the interface?

Reply:
Well I guess the first maybe is not `msg string` if we allow the structured.Error - but seems the rest doesn't need to allow anything?

Reply:
i made it error because someone there are some use cases now of zap fields. but i can refactor those to the new structured api and make info() and others behave like fmt.sprint and forget about zap field support. 
regarding the second point, the first field is optional so the start index of the zap fields is not fixed.

Reply:
My question is more about why does the interface allow passing anything but then error. Can we not have type safety here? It seems like we used to have it without issue.

Reply:
we cannot statically type it for the reason above i.e. the first field is optional.

Reply:
So the current interface is `Info(fields ...interface{})`

Valid options are:
* Info(str)
* Info(str, zap.Field, zap.Field, ...)
* Info(structured.Error, zp.Field, zap.Field, ..)

Is that accurate?

if so, can't the interface be:

`Info(msgOrError interface{}, fields ...zapcore.Field)`

Reply:
the other valid option is Info(structured.Error, str, zp.Field, zap.Field, ..)

Reply:
how about the option of removing the zapcore fields altogether? there are not many uses and the structured logging in scope is what we want people to use.

Reply:
100% fine with that